gemrb/GUIScripts/LUCommon.py

```diff
@@ -293,8 +300,11 @@ def SetupHP (pc, Level=None, LevelDiff=None):
 		#we only get the hp bonus if the old class is reactivated
 		if (Levels[0]<=Levels[1]):
 			return
-		Class = GUICommon.GetClassRowName(Dual[2], "index")
-		Class = [CommonTables.Classes.GetValue (Class, "ID")]
+		if Dual[0] == 3:
+			Class = [CommonTables.KitList.GetValue (Dual[2], 7)]
+		else:
+			ClassRow = GUICommon.GetClassRowName(Dual[2], "index")
+			Class = [CommonTables.Classes.GetValue (ClassRow, "ID")]
```

Comment:
Yes, much of 899fb8b could benefit from refactoring, since there's a lot of code like
```
if kit:
    action
else: #class
    similar action but phrased differently
```
But I guess I was feeling hacky when I wrote it.